### ISSUE TITLE
About test results 

### ISSUE THREAD

**Issue by jwyang** [peteanderson80/bottom-up-attention#4]
Hi, I just run the test code using your trained resnet101 model on the test set. I got the following numbers on object detection task:

Mean AP = 0.0146
Weighted Mean AP = 0.1799
Mean Detection Threshold = 0.328

The mean AP (1.46%) is far from the number (10.2%) you reported in the table at the bottom of readme. The weighted mean AP is a bit higher than the number you reported. I am wondering whether there is a typo in your table.

thanks!

**Comment by peteanderson80** [comment]
The mean and weighted mean numbers should be much closer than your results - the only difference is correction for class imbalance. 
Are you still having an issue with this?

**Comment by jwyang** [comment]
Hi, Peter,

I checked the evaluation code again. The mean AP is computed by averaging over all 1600 entries in app, that is:

```
print('Mean AP = {:.4f}'.format(np.mean(aps)))
```

and the weighted mean AP is computed via:

```
print('Weighted Mean AP = {:.4f}'.format(np.average(aps, weights=weights)))
```

Since there are merely a part of 1600 categories that appear in the test set (231 from my running), aps will have many zeros in it. In this case, mean(aps) should be low with no doubt.

I guess you reported the mean AP by ruling out all categories with npos = 0, and then get the average on those non-zero entries? When I did like this, I got 10.11%. It is very close to your reported numbers.

**Comment by yuzcccc** [comment]
Hi, Peter,
I am running the training scripts myself (with fewer gpus). What's number of the final training loss at iteration 380K when you trained the model? . If possible, could you please draw a training curve or provide the training log file? Thanks a lot!

**Comment by peteanderson80** [comment]
Hi @jwyang,

Sorry I haven't responded sooner. We did not exclude zeros in our calculation. It seems like there is some difference in the validation set that is being used, because our [5000 image validation set](https://github.com/peteanderson80/bottom-up-attention/blob/a8f4c8a0f15937553b41ee5d68827936bc473e4f/data/genome/val.txt) resulted in no categories with `npos = 0` in our evaluation.  

Maybe something went wrong with the dataset preprocessing? To help compare I've added the [eval.log](https://github.com/peteanderson80/bottom-up-attention/tree/7e0e4ce710b2d74264a3afe2460a12fcfbe64a7c/experiments/logs) file from our evaluation to the repo. If it helps I can also add our preprocessed `data/cache/vg_1600-400-20_val_gt_roidb.pkl` file.

**Comment by jwyang** [comment]
Hi, @peteanderson80 ,

thanks a lot for your replying, and sharing the log file. Yeah, it is very weird to me. I compared the 5000 validation images, they are the same. I will re-pull your code and re-generate the xml files to see whether I can get the same number as yours. I will let you know when I get the results.

thanks again for your help!

**Comment by peteanderson80** [comment]
Hi @yuzcccc,

I don't have the original log file, but I've added an [example log file](https://github.com/peteanderson80/bottom-up-attention/blob/09f8384031dc5ec145a18973a22cb58b5dc0e311/experiments/logs/faster_rcnn_end2end_final_ResNet-101_.txt.2017-06-29_16-27-08) from training with a single gpu for 16K iterations, which should give some indication of the expected training loss. From memory I think the final training loss was around 4.0 (compared to about 4.8 in the example log file at iteration 16300).

**Comment by peteanderson80** [comment]
Thanks @jwyang for investigating. I have shared our pickled datasets so you can see if you get the same:
- [vg_1600-400-20_train_gt_roidb.pkl](https://storage.googleapis.com/bottom-up-attention/vg_1600-400-20_train_gt_roidb.pkl)
- [vg_1600-400-20_val_gt_roidb.pkl](https://storage.googleapis.com/bottom-up-attention/vg_1600-400-20_val_gt_roidb.pkl)
